packages/react/src/components/TearSheet/TearSheetWrapper.jsx

```diff
@@ -94,7 +98,8 @@ const TearSheetWrapper = ({ isOpen, className, onCloseAllTearSheets, children })
           <div
             id={`container-${idx}`}
             key={`container-${idx}`}
-            data-testid={`container-${idx}`}
+            // TODO: use only testId in v3.
+            data-testid={testId ? `${testId}-container-${idx}` : `container-${idx}`}
```

Comment:
testId defaults to empty, so it'll fallback to the current default `container-${idx}` in v3 we'll replace this to use the passed testId